lib/dor/text_extraction/file_fetcher.rb

```diff
@@ -55,15 +55,17 @@ def write_file_with_retries(filename:, location:, max_tries: 3)
       # fetch a file from preservation and send to cloud endpoint
       def fetch_and_send_file_to_s3(filename:, s3_object:)
         logger.info("fetching #{filename} for #{druid} and sending to #{s3_object.bucket_name}")
-        s3_object.upload_stream do |upload_stream|
-          preservation_client.objects.content(
-            druid:,
-            filepath: filename,
-            on_data: proc { |data, _count| upload_stream.write(data) }
-          )
-        end
+        Thread.new do
+          s3_object.upload_stream do |upload_stream|
+            preservation_client.objects.content(
+              druid:,
+              filepath: filename,
+              on_data: proc { |data, _count| upload_stream.write(data) }
+            )
+          end
+        end.join
```

Comment:
keep the stream alive ? trying to deal with https://app.honeybadger.io/projects/52894/faults/113690502